compiler/commands.nim

```diff
@@ -312,7 +312,6 @@ proc testCompileOption*(conf: ConfigRef; switch: string, info: TLineInfo): bool
   of "symbolfiles": result = conf.symbolFiles != disabledSf
   of "genscript": result = contains(conf.globalOptions, optGenScript)
   of "threads": result = contains(conf.globalOptions, optThreads)
```

Comment:
```suggestion
  of "threads": result = contains(conf.globalOptions, optThreads)
  of "taintmode": result = false
```

so that existing code using `when compileOption("taintmode")` keeps compiling

future PR welcome to introduce a warning here:
```nim
  of "taintmode":
    issue a warning: deprecated command line option: taintmode
    result = false
```